### PR TITLE
Add box2d/2.3.1

### PR DESCRIPTION
**Pull request by danimtb**
Specify library name and version:  **box2d/2.3.1**

From https://github.com/conan-community/conan-box2d

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/box2d/all/test_package/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        cmake.configure()
+        cmake.build()
+
+    def imports(self):
+        self.copy(pattern="*.dll", src="bin", dst="bin")
+        self.copy(pattern="*.dylib", src="lib", dst="bin")
+
+    def test(self):
+        bin_path = os.path.join("bin", "test_package")
+        self.run(bin_path, run_environment=True)
```
Comment (madebr) [inline recipes/box2d/all/test_package/conanfile.py:16]:
```suggestion
        if not tools.cross_building(self.settings):
            bin_path = os.path.join("bin", "test_package")
            self.run(bin_path, run_environment=True)
```

File: recipes/box2d/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        cmake.definitions["BOX2D_BUILD_STATIC"] = not self.options.shared
+        if self.settings.os == "Windows" and self.options.shared:
+            cmake.definitions["CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS"] = True
+        cmake.definitions["BOX2D_BUILD_EXAMPLES"] = False
+        cmake.configure()
+        cmake.build()
+
+    def package(self):
+        self.copy("License.txt", dst="licenses", src="%s/Box2D" % self.source_subfolder)
+        self.copy("*.h", dst="include/Box2D", src="%s/Box2D/Box2D" % self.source_subfolder)
```
Comment (madebr) [inline recipes/box2d/all/conanfile.py:42]:
```suggestion
        self.copy("License.txt", dst="licenses", src=os.path.join(self._source_subfolder, "Box2D"))
        self.copy("*.h", dst=os.path.join("include", "Box2D"), src=os.path.join(self._source_subfolder, "Box2D", "Box2D"))
```

File: recipes/box2d/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+class Box2dConan(ConanFile):
+    name = "box2d"
+    license = "Zlib"
+    description = "Box2D is a 2D physics engine for games"
+    homepage = "http://box2d.org/"
+    url = "https://github.com/conan-io/conan-center-index"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False],
+               "fPIC": [True, False]}
+    default_options = "shared=False", "fPIC=True"
```
Comment (madebr) [inline recipes/box2d/all/conanfile.py:14]:
```suggestion
    default_options = {"shared": False", "fPIC": True",}
```

File: recipes/box2d/all/test_package/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import os
+from conans import ConanFile, CMake, tools
+
+
+class Box2DTestConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch", "cppstd"
```
Comment (madebr) [inline recipes/box2d/all/test_package/conanfile.py:6]:
```suggestion
    settings = "os", "compiler", "build_type", "arch"
```

File: recipes/box2d/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+class Box2dConan(ConanFile):
+    name = "box2d"
+    license = "Zlib"
+    description = "Box2D is a 2D physics engine for games"
+    homepage = "http://box2d.org/"
+    url = "https://github.com/conan-io/conan-center-index"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False],
+               "fPIC": [True, False]}
+    default_options = {"shared": False", "fPIC": True",}
```
Comment (madebr) [inline recipes/box2d/all/conanfile.py:14]: My bad
```suggestion
    default_options = {"shared": False", "fPIC": True,}
```

File: recipes/box2d/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+class Box2dConan(ConanFile):
+    name = "box2d"
+    license = "Zlib"
+    description = "Box2D is a 2D physics engine for games"
+    homepage = "http://box2d.org/"
+    url = "https://github.com/conan-io/conan-center-index"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False],
+               "fPIC": [True, False]}
+    default_options = {"shared": False", "fPIC": True,}
```
Comment (madebr) [inline recipes/box2d/all/conanfile.py:14]:
```suggestion
    default_options = {"shared": False, "fPIC": True,}
```

File: recipes/box2d/all/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    def source_subfolder(self):
+        return "sources"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("Box2D-%s" % self.version, self.source_subfolder)
```
Comment (madebr) [inline recipes/box2d/all/conanfile.py:28]:
```suggestion
        os.rename("box2d-%s" % self.version, self.source_subfolder)
```

File: recipes/box2d/all/conanfile.py [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+    url = "https://github.com/conan-io/conan-center-index"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False],
+               "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True,}
+    generators = "cmake"
+    exports_sources = "CMakeLists.txt"
+
+    @property
+    def source_subfolder(self):
```
Comment (madebr) [inline recipes/box2d/all/conanfile.py:19]:
```suggestion
    def _source_subfolder(self):
```

File: recipes/box2d/all/conanfile.py [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+        return "sources"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("box2d-%s" % self.version, self.source_subfolder)
```
Comment (madebr) [inline recipes/box2d/all/conanfile.py:28]:
```suggestion
        os.rename("box2d-%s" % self.version, self._source_subfolder)
```

File: recipes/box2d/all/conanfile.py [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+    generators = "cmake"
+    exports_sources = "CMakeLists.txt"
+
+    @property
+    def _source_subfolder(self):
+        return "sources"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
```
Comment (jgsogo) [inline recipes/box2d/all/conanfile.py:24]: ...or shared

Comment (danimtb) [reply recipes/box2d/all/conanfile.py:24]: right! done 😄

**Pull request merged**